frontend/src/utils/validation.js

```diff
@@ -1,26 +1,50 @@
-import {DateTimeFormatterBuilder, LocalDate, ResolverStyle} from '@js-joda/core';
-import {groupBy, isPlainObject} from 'lodash';
-
-const datePatternWithSlash = (new DateTimeFormatterBuilder).appendPattern('uuuu/MM/dd').toFormatter(ResolverStyle.STRICT);
+import {
+  DateTimeFormatterBuilder,
+  LocalDate,
+  ResolverStyle,
+} from '@js-joda/core';
+import { groupBy, isPlainObject } from 'lodash';
+
+const datePatternWithSlash = new DateTimeFormatterBuilder()
+  .appendPattern('uuuu/MM/dd')
+  .toFormatter(ResolverStyle.STRICT);
 export function checkDigit(pen) {
   const parsedPen = parseInt(pen);
-  if(parsedPen === 0 || isNaN(parsedPen)){
+  if (parsedPen === 0 || isNaN(parsedPen)) {
     return false;
   }
   const penDigits = [];
 
-  for(let i = 0; i < pen.length; i++) {
+  for (let i = 0; i < pen.length; i++) {
     penDigits[i] = parseInt(pen.charAt(i), 10);
   }
-  const S1 = penDigits.slice(0,-1).filter((element,index) => {return index % 2 === 0;}).reduce((a,b) => a+b,0);
-  const A = parseInt(penDigits.slice(0,-1).filter((element,index) => {return index % 2 === 1;}).join(''), 10);
+  const S1 = penDigits
```

Comment:
Can we use a little meaningful naming here? The SonarQube basic rules set always recommend a variable name of more than three characters.